packages/botonic-core/src/types/router.d.ts

```diff
@@ -0,0 +1,57 @@
+import { Input, Route, RouteMatcher, Routes, Session } from '.'
+import { RouteInspector } from './debug'
+
+export declare class Router {
+  constructor(routes: Routes, routeInspector?: RouteInspector)
+  routes: Routes
+  routeInspector: RouteInspector
+  processInput(
+    input: Input,
+    session?: Session,
+    lastRoutePath?: string
+  ):
+    | {
+        action: any
+        params: {
+          [x: number]: any
+        }
+        retryAction: any
+        defaultAction: any
+        lastRoutePath: any
+      }
+    | {
+        action: any
+        params: any
+        lastRoutePath: any
+        retryAction?: undefined
+        defaultAction?: undefined
+      }
+    | {
+        action: any
+        params: any
+        retryAction: any
+        lastRoutePath: any
+        defaultAction?: undefined
+      }
```

Comment:
Can these 3 type of returns be categorized into other types or interfaces? This way maybe it will be understandable why there can be 3 type of returns.

Reply:
yes, I would like to take a closer look into this in the next iteration of moving some files to TS.

Reply:
with the current code of router, it's hard to discriminate what kind of objects we can have here or if they can be optimized.